src/components/Spinner/index.tsx

```diff
@@ -1,5 +1,38 @@
 import './index.css';
 
+import { useState } from 'react';
+import { useInterval } from 'usehooks-ts';
+
 export function Spinner() {
   return <div className="lds-dual-ring" />;
 }
+
+export function ProgressSpinner({ badge }: { badge: string }) {
+  const STEP = 1;
+  const [percent, setValue] = useState(0);
+
+  useInterval(() => {
+    setValue(Math.min(percent + STEP, 100));
+  }, 200);
```

Comment:
I know this is only a temporary solution, but I don't feel super confident about this potentially infinite interval. Could you do something like this instead?
```typescript
useInterval(() => {
    setValue(Math.min(percent + STEP, 100));
  }, percent < 100 ? 200 : null);
```
So that the interval stops when `percent` reaches 100

Reply:
Yes, is temporary. But will fix that.